tensorflowonspark/pipeline.py

```diff
@@ -0,0 +1,313 @@
+from pyspark.sql import SparkSession
+
+import tensorflow as tf
+from tensorflow.contrib.saved_model.python.saved_model import reader, signature_def_utils
+from tensorflow.python.framework import ops as ops_lib
+from tensorflow.python.saved_model import loader
+from . import TFCluster, gpu_info
+
+import copy
+import logging
+import os
+import subprocess
+
+##### TensorFlowOnSpark Params
+
+class HasBatchSize(Params):
+  batch_size = Param(Params._dummy(), "batch_size", "Number of records per batch", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasBatchSize, self).__init__()
+  def setBatchSize(self, value):
+    return self._set(batch_size=value)
+  def getBatchSize(self):
+    return self.getOrDefault(self.batch_size)
+
+class HasClusterSize(Params):
+  cluster_size = Param(Params._dummy(), "cluster_size", "Number of nodes in the cluster", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasClusterSize, self).__init__()
+  def setClusterSize(self, value):
+    return self._set(cluster_size=value)
+  def getClusterSize(self):
+    return self.getOrDefault(self.cluster_size)
+
+class HasEpochs(Params):
+  epochs = Param(Params._dummy(), "epochs", "Number of epochs to train", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasEpochs, self).__init__()
+  def setEpochs(self, value):
+    return self._set(epochs=value)
+  def getEpochs(self):
+    return self.getOrDefault(self.epochs)
+
+class HasInputTensor(Params):
+  tensor_in = Param(Params._dummy(), "tensor_in", "Name of input tensor in signature def", typeConverter=TypeConverters.toString)
+  def __init__(self):
+    super(HasInputTensor, self).__init__()
+  def setInputTensor(self, value):
+    return self._set(tensor_in=value)
+  def getInputTensor(self):
+    return self.getOrDefault(self.tensor_in)
+
+class HasModelDir(Params):
+  model_dir = Param(Params._dummy(), "model_dir", "Path to save/load model checkpoints", typeConverter=TypeConverters.toString)
+  def __init__(self):
+    super(HasModelDir, self).__init__()
+  def setModelDir(self, value):
+    return self._set(model_dir=value)
+  def getModelDir(self):
+    return self.getOrDefault(self.model_dir)
+
+class HasNumPS(Params):
+  num_ps = Param(Params._dummy(), "num_ps", "Number of PS nodes in cluster", typeConverter=TypeConverters.toInt)
+  def __init__(self):
+    super(HasNumPS, self).__init__()
+  def setNumPS(self, value):
+    return self._set(num_ps=value)
+  def getNumPS(self):
+    return self.getOrDefault(self.num_ps)
+
+class HasOutputTensor(Params):
+  tensor_out = Param(Params._dummy(), "tensor_out", "Name of output tensor in signature def", typeConverter=TypeConverters.toString)
+  def __init__(self):
+    super(HasOutputTensor, self).__init__()
+  def setOutputTensor(self, value):
+    return self._set(tensor_out=value)
+  def getOutputTensor(self):
+    return self.getOrDefault(self.tensor_out)
+
+class HasRDMA(Params):
```

Comment:
all the params make sense to me, but i'm not sure about this one. it seems too TFoS specific (correct me if I'm wrong since I don't have that much context). how crucial is it that the user be able to set this?